ParserTests/CommentsTests.cs

```diff
@@ -235,4 +461,618 @@ public void TestMixedEOLComment()
+
+            var dump = lexer.ToString();
+            var code = "1\n2\r\n/* multi line \rcomment on 2 lines */ 3.0";
+            List<Token<CommentsTokenAlternative>> tokens = null;
+            tokens = lexer.Tokenize(code).ToList();
+
+            Assert.Equal(5, tokens.Count);
+
+            var token1 = tokens[0];
+            var token2 = tokens[1];
+            var token3 = tokens[2];
+            var token4 = tokens[3];
+
+            Assert.Equal(CommentsTokenAlternative.INT, token1.TokenID);
+            Assert.Equal("1", token1.Value);
+            Assert.Equal(0, token1.Position.Line);
+            Assert.Equal(0, token1.Position.Column);
+
+            Assert.Equal(CommentsTokenAlternative.INT, token2.TokenID);
+            Assert.Equal("2", token2.Value);
+            Assert.Equal(1, token2.Position.Line);
+            Assert.Equal(0, token2.Position.Column);
+            Assert.Equal(CommentsTokenAlternative.COMMENT, token3.TokenID);
+            Assert.Equal(" multi line \rcomment on 2 lines ", token3.Value);
+            Assert.Equal(2, token3.Position.Line);
+            Assert.Equal(0, token3.Position.Column);
+            Assert.Equal(CommentsTokenAlternative.DOUBLE, token4.TokenID);
+            Assert.Equal("3.0", token4.Value);
+            Assert.Equal(3, token4.Position.Line);
+            Assert.Equal(22, token4.Position.Column);
+        }
+    }
+
+    public class SingleLineCommentsTest
+    {
+        [Fact]
+        public void NotEndingMultiComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<SingleLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<SingleLineCommentsToken>;
+
+
+            var dump = lexer.ToString();
+
+            var code = @"1
+2 /* not ending
+comment";
+
+            var tokens = lexer.Tokenize(code).ToList();
+
+            Assert.Equal(3, tokens.Count);
+
+            var token1 = tokens[0];
+            var token2 = tokens[1];
+
+            Assert.Equal(SingleLineCommentsToken.INT, token1.TokenID);
+            Assert.Equal("1", token1.Value);
+            Assert.Equal(0, token1.Position.Line);
+            Assert.Equal(0, token1.Position.Column);
+
+            Assert.Equal(SingleLineCommentsToken.INT, token2.TokenID);
+            Assert.Equal("2", token2.Value);
+            Assert.Equal(1, token2.Position.Line);
+            Assert.Equal(0, token2.Position.Column);
+        }
+
+        [Fact]
+        public void TestGenericMultiLineComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<SingleLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<SingleLineCommentsToken>;
+
+
+            var dump = lexer.ToString();
+
+            var code = @"1
+2 /* multi line 
+comment on 2 lines */ 3.0";
```

Comment:
How should the parser behave here? Right now it ignores the comment as well as the DOUBLE (3.0), generating two tokens for the INT and one empty token at the end.

I would expect the comments and text to be completely ignored and the INT 2 and DOUBLE 3.0 be tokenized or to completely fail compiling with an error. The actual behavior seems erroneous for me.

Reply:
Let's clarify : this UT is using SingleLineCommentsToken so multiline comment should not be valid token and hence lexing should fail. This could explain the fact that 3.0 is ignored but the right behavior is a lexing error (unexpected token "/" at line 2 column 3 I guess).